cfme/containers/topology.py

```diff
@@ -1,23 +1,33 @@
 # -*- coding: utf-8 -*-
+import attr
+
 from navmazing import NavigateToAttribute
 
+from cfme.modeling.base import BaseCollection, BaseEntity
 from cfme.common import TopologyMixin
 from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
-from cfme.utils.appliance import Navigatable
 
 
-class Topology(TopologyMixin, Navigatable):
+class Topology(TopologyMixin, BaseEntity):
 
-    def __init__(self, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
+    region = attr.ib(default=None)
+    provider = attr.ib(default=None)
+    datastore = attr.ib(default=None)
```

Comment:
Question:
why region, provider and datastore?

Reply:
Oh that's a mistake. I was following this other collection. Will change it.